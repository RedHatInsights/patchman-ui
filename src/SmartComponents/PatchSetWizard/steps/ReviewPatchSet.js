import React from 'react';
import {
    Text,
    TextContent,
    Stack,
    StackItem,
    TextVariants,
    TextList,
    TextListVariants,
    TextListItem,
    TextListItemVariants
} from '@patternfly/react-core';
import useFormApi from '@data-driven-forms/react-form-renderer/use-form-api';

const ReviewPatchSet = () => {
    const formOptions = useFormApi();
    const { values } = formOptions.getState();
    const { name, description, toDate } = values.existing_patch_set || values;
    const { systems } = values;

    return (
        <Stack hasGutter>
            <StackItem>
                <TextContent style={{ marginTop: '-15px' }}>
                    <Text component={TextVariants.p}>
                        Review the information below and click <b>Submit</b> to complete patch set creation
                    </Text>
                </TextContent>
            </StackItem>
            <StackItem>
                <TextContent>
                    <TextList component={TextListVariants.dl}>
                        <TextListItem component={TextListItemVariants.dt}>Name:</TextListItem>
                        <TextListItem component={TextListItemVariants.dd}>{name}</TextListItem>
                        <TextListItem component={TextListItemVariants.dt}>Description:</TextListItem>
                        <TextListItem component={TextListItemVariants.dd}>{description}</TextListItem>
                        <TextListItem component={TextListItemVariants.dt}>Date:</TextListItem>
                        <TextListItem component={TextListItemVariants.dd}>{toDate}</TextListItem>
                        <TextListItem component={TextListItemVariants.dt}>Selected systems:</TextListItem>
                        <TextListItem component={TextListItemVariants.dd}>
                            {systems && Object.values(systems).filter(system => system).length}
                        </TextListItem>
                    </TextList>
                </TextContent>
            </StackItem>
        </Stack>
    );
};

export default ReviewPatchSet;
