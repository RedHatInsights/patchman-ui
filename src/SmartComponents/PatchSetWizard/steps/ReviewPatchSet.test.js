import ReviewPatchSet from './ReviewPatchSet';
import toJson from 'enzyme-to-json';
import useFormApi from '@data-driven-forms/react-form-renderer/use-form-api';

jest.mock('@data-driven-forms/react-form-renderer/use-form-api',
    () => ({
        __esModule: true,
        default: jest.fn()
    })
);

describe('ReviewPatchSet.js', () => {
    it('Should display all Patch Template configuration', () => {
        useFormApi.mockReturnValue(({
            getState: () => ({
                values: {
                    name: 'test-name',
                    description: 'test-description',
                    toDate: '2022-06-14',
                    systems: {
                        'test-system-id-1': true,
                        'test-system-id-2': true
                    }
                }
            })
        }));
        const wrapper = mount(<ReviewPatchSet />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should display available configuration fields only', () => {
        useFormApi.mockReturnValue(({
            getState: () => ({
                values: {
                    name: 'test-name',
                    systems: {
                        'test-system-id-1': true,
                        'test-system-id-2': true
                    }
                }
            })
        }));
        const wrapper = mount(<ReviewPatchSet/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

