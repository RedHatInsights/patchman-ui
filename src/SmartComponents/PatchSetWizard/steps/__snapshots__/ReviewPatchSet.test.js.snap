// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewPatchSet.js Should display all Patch Template configuration 1`] = `
<ReviewPatchSet>
  <Stack
    hasGutter={true}
  >
    <div
      className="pf-l-stack pf-m-gutter"
    >
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <TextContent>
            <div
              className="pf-c-content"
            >
              <Text
                component="h2"
              >
                <h2
                  className=""
                  data-ouia-component-id="OUIA-Generated-Text-1"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe={true}
                  data-pf-content={true}
                >
                  Review 
                </h2>
              </Text>
            </div>
          </TextContent>
        </div>
      </StackItem>
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <TextContent
            style={
              Object {
                "marginTop": "-15px",
              }
            }
          >
            <div
              className="pf-c-content"
              style={
                Object {
                  "marginTop": "-15px",
                }
              }
            >
              <Text
                component="p"
              >
                <p
                  className=""
                  data-ouia-component-id="OUIA-Generated-Text-2"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe={true}
                  data-pf-content={true}
                >
                  Review the details of your template.
                </p>
              </Text>
            </div>
          </TextContent>
        </div>
      </StackItem>
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <TextContent>
            <div
              className="pf-c-content"
            >
              <Text
                className="pf-u-mt-md pf-u-mb-sm"
                component="h2"
              >
                <h2
                  className="pf-u-mt-md pf-u-mb-sm"
                  data-ouia-component-id="OUIA-Generated-Text-3"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe={true}
                  data-pf-content={true}
                >
                  Content
                </h2>
              </Text>
              <TextList
                component="dl"
              >
                <dl
                  className=""
                >
                  <TextListItem
                    component="dt"
                    style={
                      Object {
                        "minWidth": 220,
                      }
                    }
                  >
                    <dt
                      className=""
                      style={
                        Object {
                          "minWidth": 220,
                        }
                      }
                    >
                      Red Hat advisories up to date
                    </dt>
                  </TextListItem>
                  <TextListItem
                    component="dd"
                  >
                    <dd
                      className=""
                    >
                      14 June 2022
                    </dd>
                  </TextListItem>
                </dl>
              </TextList>
            </div>
          </TextContent>
        </div>
      </StackItem>
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <TextContent>
            <div
              className="pf-c-content"
            >
              <Text
                className="pf-u-mt-md pf-u-mb-sm"
                component="h2"
              >
                <h2
                  className="pf-u-mt-md pf-u-mb-sm"
                  data-ouia-component-id="OUIA-Generated-Text-4"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe={true}
                  data-pf-content={true}
                >
                  Details
                </h2>
              </Text>
              <TextList
                component="dl"
              >
                <dl
                  className=""
                >
                  <TextListItem
                    component="dt"
                    style={
                      Object {
                        "minWidth": 220,
                      }
                    }
                  >
                    <dt
                      className=""
                      style={
                        Object {
                          "minWidth": 220,
                        }
                      }
                    >
                      Name
                    </dt>
                  </TextListItem>
                  <TextListItem
                    component="dd"
                  >
                    <dd
                      className=""
                    >
                      test-name
                    </dd>
                  </TextListItem>
                  <TextListItem
                    component="dt"
                    style={
                      Object {
                        "minWidth": 220,
                      }
                    }
                  >
                    <dt
                      className=""
                      style={
                        Object {
                          "minWidth": 220,
                        }
                      }
                    >
                      Description
                    </dt>
                  </TextListItem>
                  <TextListItem
                    component="dd"
                  >
                    <dd
                      className=""
                    >
                      test-description
                    </dd>
                  </TextListItem>
                </dl>
              </TextList>
            </div>
          </TextContent>
        </div>
      </StackItem>
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <TextContent>
            <div
              className="pf-c-content"
            >
              <Text
                className="pf-u-mt-md pf-u-mb-sm"
                component="h2"
              >
                <h2
                  className="pf-u-mt-md pf-u-mb-sm"
                  data-ouia-component-id="OUIA-Generated-Text-5"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe={true}
                  data-pf-content={true}
                >
                  Systems
                </h2>
              </Text>
              <TextList
                component="dl"
              >
                <dl
                  className=""
                >
                  <TextListItem
                    component="dt"
                    style={
                      Object {
                        "minWidth": 220,
                      }
                    }
                  >
                    <dt
                      className=""
                      style={
                        Object {
                          "minWidth": 220,
                        }
                      }
                    >
                      Template applied to
                    </dt>
                  </TextListItem>
                  <TextListItem
                    component="dd"
                  >
                    <dd
                      className=""
                    >
                       2 systems 
                    </dd>
                  </TextListItem>
                </dl>
              </TextList>
            </div>
          </TextContent>
        </div>
      </StackItem>
    </div>
  </Stack>
</ReviewPatchSet>
`;

exports[`ReviewPatchSet.js Should display available configuration fields only 1`] = `
<ReviewPatchSet>
  <Stack
    hasGutter={true}
  >
    <div
      className="pf-l-stack pf-m-gutter"
    >
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <TextContent>
            <div
              className="pf-c-content"
            >
              <Text
                component="h2"
              >
                <h2
                  className=""
                  data-ouia-component-id="OUIA-Generated-Text-6"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe={true}
                  data-pf-content={true}
                >
                  Review 
                </h2>
              </Text>
            </div>
          </TextContent>
        </div>
      </StackItem>
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <TextContent
            style={
              Object {
                "marginTop": "-15px",
              }
            }
          >
            <div
              className="pf-c-content"
              style={
                Object {
                  "marginTop": "-15px",
                }
              }
            >
              <Text
                component="p"
              >
                <p
                  className=""
                  data-ouia-component-id="OUIA-Generated-Text-7"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe={true}
                  data-pf-content={true}
                >
                  Review the details of your template.
                </p>
              </Text>
            </div>
          </TextContent>
        </div>
      </StackItem>
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <TextContent>
            <div
              className="pf-c-content"
            >
              <Text
                className="pf-u-mt-md pf-u-mb-sm"
                component="h2"
              >
                <h2
                  className="pf-u-mt-md pf-u-mb-sm"
                  data-ouia-component-id="OUIA-Generated-Text-8"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe={true}
                  data-pf-content={true}
                >
                  Content
                </h2>
              </Text>
              <TextList
                component="dl"
              >
                <dl
                  className=""
                >
                  <TextListItem
                    component="dt"
                    style={
                      Object {
                        "minWidth": 220,
                      }
                    }
                  >
                    <dt
                      className=""
                      style={
                        Object {
                          "minWidth": 220,
                        }
                      }
                    >
                      Red Hat advisories up to date
                    </dt>
                  </TextListItem>
                  <TextListItem
                    component="dd"
                  >
                    <dd
                      className=""
                    >
                      N/A
                    </dd>
                  </TextListItem>
                </dl>
              </TextList>
            </div>
          </TextContent>
        </div>
      </StackItem>
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <TextContent>
            <div
              className="pf-c-content"
            >
              <Text
                className="pf-u-mt-md pf-u-mb-sm"
                component="h2"
              >
                <h2
                  className="pf-u-mt-md pf-u-mb-sm"
                  data-ouia-component-id="OUIA-Generated-Text-9"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe={true}
                  data-pf-content={true}
                >
                  Details
                </h2>
              </Text>
              <TextList
                component="dl"
              >
                <dl
                  className=""
                >
                  <TextListItem
                    component="dt"
                    style={
                      Object {
                        "minWidth": 220,
                      }
                    }
                  >
                    <dt
                      className=""
                      style={
                        Object {
                          "minWidth": 220,
                        }
                      }
                    >
                      Name
                    </dt>
                  </TextListItem>
                  <TextListItem
                    component="dd"
                  >
                    <dd
                      className=""
                    >
                      test-name
                    </dd>
                  </TextListItem>
                  <TextListItem
                    component="dt"
                    style={
                      Object {
                        "minWidth": 220,
                      }
                    }
                  >
                    <dt
                      className=""
                      style={
                        Object {
                          "minWidth": 220,
                        }
                      }
                    >
                      Description
                    </dt>
                  </TextListItem>
                  <TextListItem
                    component="dd"
                  >
                    <dd
                      className=""
                    >
                      No description provided
                    </dd>
                  </TextListItem>
                </dl>
              </TextList>
            </div>
          </TextContent>
        </div>
      </StackItem>
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <TextContent>
            <div
              className="pf-c-content"
            >
              <Text
                className="pf-u-mt-md pf-u-mb-sm"
                component="h2"
              >
                <h2
                  className="pf-u-mt-md pf-u-mb-sm"
                  data-ouia-component-id="OUIA-Generated-Text-10"
                  data-ouia-component-type="PF4/Text"
                  data-ouia-safe={true}
                  data-pf-content={true}
                >
                  Systems
                </h2>
              </Text>
              <TextList
                component="dl"
              >
                <dl
                  className=""
                >
                  <TextListItem
                    component="dt"
                    style={
                      Object {
                        "minWidth": 220,
                      }
                    }
                  >
                    <dt
                      className=""
                      style={
                        Object {
                          "minWidth": 220,
                        }
                      }
                    >
                      Template applied to
                    </dt>
                  </TextListItem>
                  <TextListItem
                    component="dd"
                  >
                    <dd
                      className=""
                    >
                       2 systems 
                    </dd>
                  </TextListItem>
                </dl>
              </TextList>
            </div>
          </TextContent>
        </div>
      </StackItem>
    </div>
  </Stack>
</ReviewPatchSet>
`;
